compiler/src/dotty/tools/dotc/transform/localopt/StringInterpolatorOpt.scala

```diff
@@ -0,0 +1,59 @@
+package dotty.tools.dotc.transform.localopt
+
+import dotty.tools.dotc.ast.Trees._
+import dotty.tools.dotc.ast.tpd
+import dotty.tools.dotc.core.Constants.Constant
+import dotty.tools.dotc.core.Contexts.Context
+import dotty.tools.dotc.core.StdNames._
+import dotty.tools.dotc.core.Symbols._
+import dotty.tools.dotc.transform.MegaPhase.MiniPhase
+
+/**
+  * Created by wojtekswiderski on 2018-01-24.
+  */
+class StringInterpolatorOpt extends MiniPhase {
+  import tpd._
+
+  override def phaseName: String = "stringInterpolatorOpt"
+
+  private object StringContextIntrinsic {
+    def unapply(tree: Apply)(implicit ctx: Context): Option[(List[Tree], List[Tree])] = {
+      tree match {
+        case Apply(Select(Apply(Select(Ident(nme.StringContext), nme.apply),
+                   List(SeqLiteral(strs, _))), id), List(SeqLiteral(elems, _))) =>
+          if (id == nme.raw_) Some(strs, elems)
+          else if (id == nme.s) {
+            try {
+              val escapedStrs = strs.mapConserve { str =>
+                val strValue = str.asInstanceOf[Literal].const.stringValue
```

Comment:
I don't think you can assume that `str` is always a constant literal. One could write:
```scala
val foo = "Hello"
StringContext(foo).s()
```